riscv-c-api.md

```diff
@@ -58,19 +58,33 @@ For example:
-| __riscv_c               | Arch Version | `C` extension is available.   |
-| __riscv_b               | Arch Version | `B` extension is available.   |
-| __riscv_v               | Arch Version | `V` extension is available.   |
-| __riscv_zfh             | Arch Version | `Zfh` extension is available. |
+
+| Name                    | Value        | When defined                                                 |
+| ----------------------- | ------------ | ------------------------------------------------------------ |
+| __riscv_arch_test       | 1            | Compiler supports architecture extension test macros.        |
+| __riscv_i               | Arch Version | `I` extension is available.                                  |
+| __riscv_e               | Arch Version | `E` extension is available.                                  |
+| __riscv_m               | Arch Version | `M` extension is available.                                  |
+| __riscv_a               | Arch Version | `A` extension is available.                                  |
+| __riscv_f               | Arch Version | `F` extension is available.                                  |
+| __riscv_d               | Arch Version | `D` extension is available.                                  |
+| __riscv_c               | Arch Version | `C` extension is available.                                  |
+| __riscv_b               | Arch Version | `B` extension is available.                                  |
+| __riscv_v               | Arch Version | `V` extension is available.                                  |
+| __riscv_zfh             | Arch Version | `Zfh` extension is available.                                |
+| __riscv_zk              | Arch Version | `Zbkb` `Zbkc` `Zbkx` `Zkne` `Zknd` `Zknh` are all available. |
+| __riscv_zkn             | Arch Version | `Zbkb` `Zbkc` `Zbkx` `Zkne` `Zknd` `Zknh` are all available. |
+| __riscv_zks             | Arch Version | `Zbkb` `Zbkc` `Zbkx` `Zksed` `Zksh` are all available.       |
+| __riscv_zbkb            | Arch Version | `Zbkb` extension is available.                               |
+| __riscv_zbkc            | Arch Version | `Zbkc` extension is available.                               |
+| __riscv_zbkx            | Arch Version | `Zbkx` extension is available.                               |
+| __riscv_zknd            | Arch Version | `Zknd` extension is available.                               |
+| __riscv_zkne            | Arch Version | `Zkne` extension is available.                               |
+| __riscv_zknh            | Arch Version | `Zknh` extension is available.                               |
+| __riscv_zksed           | Arch Version | `Zksed` extension is available.                              |
+| __riscv_zksh            | Arch Version | `Zksh` extension is available.                               |
+| __riscv_zkt             | Arch Version | Target asserts `Zkt` (data-independent latency extension).   |
```

Comment:
Is Zkr missing?

Reply:
Yes. The compiler wouldn't necessarily know if it is there or not, especially at compile time. Zkr does not introduce new instructions, but a CSR (called "seed", and having address 0x015), so it should be accessed with whatever builtin or intrinsic is used to issue the `csrrw rd, 0x015, x0` instruction. See test code at https://github.com/rvkrypto/rvkrypto-fips/blob/main/test/test_zkr.c

In a Linux system, under a typical configuration, the CSR is only available to the hypervisor or the Kernel, and hence it is of limited general programming interest.

Reply:
It looks like the LLVM implementation accepts Zkr in -march. And Zk in -march implies Zkr. This causes the __riscv_zkr to be emitted from the preprocecessor and it appears in elf attributes.

What is the desired behavior?

Reply:
Yes, thinking about it, it's better to have it -- perhaps future versions of the Zkr extension may actually define ISA extensions for entropy / random numbers, not just a CSR. So I added `__riscv_zkr` in https://github.com/mjosaarinen/riscv-c-api-doc/commit/8c6c806549faa79632991d5fc78c4abb24e040f8